index.js

```diff
@@ -1,5 +1,72 @@
+const yaml = require('js-yaml');
+
 module.exports = (robot) => {
+  robot.on('pull_request.closed', sync);
+
+  async function sync(context) {
+    if (!context.payload.pull_request.merged) {
+      // If a pull request is closed without merging,
+      // then we have no work to do.
+      return;
+    }
+
+    const config = await readConfig(context.payload.repository, context.github);
+    if (!config.perform) {
+      // If we didn't find a config file,
+      // then we have nothing to do.
+      return;
+    }
+
+    org = context.payload.repository.owner.login;
+    slug = context.payload.repository.name;
+
+    // STUCK: Uhm... what now? How do I actually save the value?
+    context.github.orgs.getTeams({ org: org, page: 1, per_page: 20}, getTeam)
```

Comment:
Ignoring that i've never tried the 2 param version of the github api methods, I'm going to assume they work as expected.

So from a high level, you can use async/await from node 7+ and ES2017 or Promises as you are attempting here.

The code as written is more async/await style, so the quick translation would be:
```javascript
let teams = await context.github.orgs.getTeams({ org: org, page: 1, per_page: 20});
let correct_team = getTeam(teams); //ish. Function def needs to be updated for that to make sense
```

When I'm doing something that lends itself typically to a promises flow (like this), my current hack is to wrap a helper function in aysnc/await, something like:

```javascript
async getTeamForName(teamName, github){
 return github.orgs.getTeams({ org: org, page: 1, per_page: 20}, getTeam);
}
```
So you can chain Promises.then.then.then stuff together when it makes sense, but then treat it like a single returnable object.
The other option, is you embrace the promise chain, and everything flows like that. You are expecting to get a team/team.id, and return to normal flow. Instead, you should be thinking about it like passing that as the input to the next then/Promise. 

It kind of makes sense if you embrace the truly sequential nature of those commands  (get All Teams, find the proper Team, Do something for that team, Ensure that thing happened for that team).

What does this all mean? Eh, you probably want to cheat with async/await. But I found this [intro to promises](https://developers.google.com/web/fundamentals/getting-started/primers/promises) really changed my view on them.